From 27a39ad89ca0c9fe416d7e19eaf2cb2fe705f305 Mon Sep 17 00:00:00 2001
From: Grzegorz Jaszczyk <jaz@semihalf.com>
Date: Fri, 16 Sep 2016 16:12:37 +0200
Subject: [PATCH 0982/2241] arm64: dts: marvell: add ioexpander description for
 Armada 8040 DB board

Provide required nodes describing both expanders found on the board. The
expander0 can be used e.g. for configuring the CP0's usb vbus CP0, while
the expander1 for CP1's usb vbus.

MPP62 is configured as gpio and it is connected to INT signal of both
expanders.

Change-Id: I5286250bb12fb0a937016b3f491d20c5d5cd393f
Signed-off-by: Grzegorz Jaszczyk <jaz@semihalf.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/32935
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
---
 arch/arm64/boot/dts/marvell/armada-8040-db.dtsi | 20 ++++++++++++++++++++
 arch/arm64/boot/dts/marvell/armada-8040.dtsi    |  4 ++++
 2 files changed, 24 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi b/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi
index 57fe8c9..62dedfc 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi
@@ -87,6 +87,26 @@
 				clock-frequency = <100000>;
 				pinctrl-names = "default";
 				pinctrl-0 = <&cpm_i2c0_pins>;
+
+				/* U31 */
+				expander0: pca9555@21 {
+					compatible = "nxp,pca9555";
+					pinctrl-names = "default";
+					pinctrl-0 = <&pca0_pins>;
+					gpio-controller;
+					#gpio-cells = <2>;
+					reg = <0x21>;
+				};
+
+				/* U25 */
+				expander1: pca9555@25 {
+					compatible = "nxp,pca9555";
+					pinctrl-names = "default";
+					gpio-controller;
+					#gpio-cells = <2>;
+					reg = <0x25>;
+				};
+
 				eeprom@57 {
 					compatible = "at24,24c64";
 					reg = <0x57>;
diff --git a/arch/arm64/boot/dts/marvell/armada-8040.dtsi b/arch/arm64/boot/dts/marvell/armada-8040.dtsi
index 26dc838..2badcbb 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-8040.dtsi
@@ -147,6 +147,10 @@
 						       "mpp53", "mpp54", "mpp55";
 					marvell,function = "ge1";
 				};
+				pca0_pins: pca0_pins {
+					marvell,pins = "mpp62";
+					marvell,function = "gpio";
+				};
 			};
 			usb3h0: usb3@500000 {
 				#stream-id-cells = <1>;
-- 
2.7.4

